ext/agent/test/Makefile

```diff
@@ -1,5 +1,25 @@
-all:
-	@# TODO
+CPP       = g++
+OBJ	  = gtest_main.o agent_unittest.o $(GTEST_LIB)
+LINKOBJ	  = gtest_main.o agent_unittest.o $(GTEST_LIB)
+LINKFLAGS = -lpthread -lgcov
+BIN_TEST  = bin
+BIN       = $(BIN_TEST)/test_agent
+CXXFLAGS  = -Wall -DCONFIG_AGENT_TEST -isystem ${GTEST_DIR}/include -isystem ${GMOCK_DIR}/include
+GCOV_FILE = *.gcno *.gcda *.gcov
+
+.PHONY : all clean
+
+all: $(BIN)
 
 clean:
-	@# TODO
+	rm -rf $(OBJ) $(BIN_DEST) $(GCOV_FILE)
+
+$(BIN): $(OBJ)
+	mkdir -p $(BIN_DEST)
+	$(CPP) $(LINKOBJ) -o $(BON) $(LIBS) $(LINKFLAGS)
+
+gtest_main.o: gtest_main.cc
+	$(CPP) -c gtest_main.cc -o gtest_main.o $(CXXFLAGS)
+
```

Comment:
No corresponding file for agent testing. Thus the compilation will failed and so do the Travis check.
Please add the basic testing files for the `agent` extension.